crates/iota-storage/tests/key_value_tests.rs

```diff
@@ -514,3 +515,57 @@ mod simtests {
         assert_eq!(result, vec![Some(ev)]);
     }
 }
+
+#[test]
+fn test_key_to_path_and_back() {
+    let tx = TransactionDigest::random();
+    let key = Key::Transaction(tx);
+    let path_elts = key.to_path_elements();
+    assert_eq!(
+        path_elements_to_key(path_elts.0, path_elts.1.as_str()).unwrap(),
+        key
+    );
+
+    let key = Key::TransactionEffects(TransactionDigest::random());
+    let path_elts = key.to_path_elements();
+    assert_eq!(
+        path_elements_to_key(path_elts.0, path_elts.1.as_str()).unwrap(),
+        key
+    );
+
```

Comment:
Just a reminder that the `Key::Events` was removed at https://github.com/iotaledger/iota/pull/5259, so we don't have the corresponding testing from the upstream here.